forms-web-app/tests/unit/controllers/appellant-submission/site-access.test.js

```diff
@@ -0,0 +1,88 @@
+const siteAccessController = require('../../../../src/controllers/appellant-submission/site-access');
+const { createOrUpdateAppeal } = require('../../../../src/lib/appeals-api-wrapper');
+const { VIEW } = require('../../../../src/lib/views');
+const logger = require('../../../../src/lib/logger');
+const { EMPTY_APPEAL } = require('../../../../src/lib/appeals-api-wrapper');
+const { mockReq, mockRes } = require('../../mocks');
+
+jest.mock('../../../../src/lib/appeals-api-wrapper');
+jest.mock('../../../../src/lib/logger');
+
+const req = mockReq();
+const res = mockRes();
+
+const sectionName = 'appealSiteSection';
+const taskName = 'siteAccess';
+
+describe('controller/appellant-submission/site-access', () => {
+  describe('getSiteAccess', () => {
+    it('should call the correct template', () => {
+      siteAccessController.getSiteAccess(req, res);
+
+      expect(res.render).toHaveBeenCalledWith(VIEW.APPELLANT_SUBMISSION.SITE_ACCESS, {
+        appeal: req.session.appeal,
+      });
+    });
+  });
+
+  describe('postSiteAccess', () => {
+    it('should re-render the template with errors if there is any validation error', async () => {
+      const mockRequest = {
+        ...req,
+        body: {
+          'site-access': 'bad value',
+          errors: { a: 'b' },
+          errorSummary: [{ text: 'There were errors here', href: '#' }],
+        },
+      };
+      await siteAccessController.postSiteAccess(mockRequest, res);
+
+      expect(res.redirect).not.toHaveBeenCalled();
+      expect(res.render).toHaveBeenCalledWith(VIEW.APPELLANT_SUBMISSION.SITE_ACCESS, {
+        appeal: req.session.appeal,
+        errorSummary: [{ text: 'There were errors here', href: '#' }],
+        errors: { a: 'b' },
+      });
+    });
+
+    it('should re-render the template with errors if there is any api call error', async () => {
+      const mockRequest = {
+        ...req,
+        body: {},
+      };
+
+      const error = new Error('Cheers');
+      createOrUpdateAppeal.mockImplementation(() => Promise.reject(error));
+
+      await siteAccessController.postSiteAccess(mockRequest, res);
+
+      expect(res.redirect).not.toHaveBeenCalled();
+      expect(logger.error).toHaveBeenCalledWith(error);
+      expect(res.render).toHaveBeenCalledWith(VIEW.APPELLANT_SUBMISSION.SITE_ACCESS, {
+        appeal: req.session.appeal,
+        errors: {},
+        errorSummary: [{ text: error.toString(), href: '#' }],
+      });
+    });
+
+    it('should redirect to `/appellant-submission/site-access-safety` if valid', async () => {
+      createOrUpdateAppeal.mockImplementation(() => JSON.stringify({ good: 'data' }));
+      const mockRequest = {
+        ...req,
+        body: {
+          'site-access': 'yes',
+        },
+      };
+      await siteAccessController.postSiteAccess(mockRequest, res);
+
+      expect(res.redirect).toHaveBeenCalledWith(`/${VIEW.APPELLANT_SUBMISSION.SITE_ACCESS_SAFETY}`);
+
+      const goodAppeal = EMPTY_APPEAL;
```

Comment:
Just FYI but by doing this you'll modifiy the imported EMPTY_APPEAL, that's why I made a deep copy with the Json parse stringify thing.

I tried to improve this in my PR so I'll change it after this merge but it might save you some debugging time one day :)  

All good for the task part !

Reply:
sorry yeah, you did say this - I had written this before asking why - will fixup

Reply:
ok, fixed and pushed. I wonder if :+1: 

`const goodAppeal = { ...EMPTY_APPEAL };`

would do the same job.